doc/nrf/libraries/others/trusted_storage.rst

```diff
@@ -0,0 +1,153 @@
+
+Architecture
+============
+
+Most microprocessors (MCUs) have on-chip non-volatile storage that can be made inaccessible except to software running on the MCU.
+If that storage is made inaccessible to software other than that of the Platform Root of Trust (PRoT), it can be used to store key material, replay protection values, or other data critical to the secure operation of the device.
+In addition, many platforms also have external storage that requires confidentiality, integrity, and replay protection from attackers with physical access to the device.
+These two storage types are reflected in the architecture of the trusted storage library.
+
+.. figure:: images/trusted_storage.svg
+   :alt: Overview of the trusted storage library architecture
+
+Interfaces
+==========
+
+The trusted storage library provides two storage interfaces for use with device-protected storage:
+
+* PSA internal trusted storage, designed to store critical data that must be placed inside internal non-volatile memory.
+  The size of the storage available by the Internal Trusted Storage API is expected to be limited, and therefore should be used for small, security-critical values.
+  Examples of assets that require this kind of storage are replay protection values for external storage and keys for use by components of the PSA Root of Trust.
+* PSA protected storage, designed to store all other critical data that do not need to be stored inside internal non-volatile memory.
+
+The two interfaces provide a consistent way for applications to access the storage types, through PSA Internal Trusted Storage API and PSA Protected Storage API.
+
+There are two backend interfaces:
+
+* ``SECURE_STORAGE_BACKEND`` that defines a backend for handling of encryption, authentication or other means of validation of the stored data.
+  It is responsible for what is happening with the data before and after they are read from or written to non-volatile storage.
+* ``SECURE_STORAGE_STORAGE_BACKEND`` that defines a backend that handles how the data are written to non-volatile storage.
+
+The trusted storage library provides ``SECURE_STORAGE_BACKEND_AEAD`` and ``SECURE_STORAGE_STORAGE_BACKEND_SETTINGS`` as default backends for the ``SECURE_STORAGE_BACKEND`` and the ``SECURE_STORAGE_STORAGE_BACKEND`` respectively.
+
+Backends
+========
+
+The following backends are used in the trusted storage library:
+
+SECURE_STORAGE_BACKEND_AEAD
+   Uses an AEAD scheme to provide integrity, confidentiality and authenticity.
+   The trusted storage library provides the ``SECURE_STORAGE_BACKEND_AEAD`` backend, but it has support for adding other secure implementation backends to provide various levels of trust, depending on the device security features.
+
+   Uses a separate backend to perform the AEAD operation, set by the Kconfig option :kconfig:option:`SECURE_STORAGE_AEAD_BACKEND_CRYPTO`.
+   The nonce and the key for the AEAD operation are provided by separate backends, allowing a custom application.
```

Comment:
```suggestion
   The nonce and the key for the AEAD operation are provided by separate backends, allowing customization 
```
Else it sounds a bit wired what custom application is this allowing? 
Alternatively `allowing a custom implementation` could be also an option.